### PR TITLE
Added support for agent host

### PR DESCRIPTION
**Pull request by sammychinedu2ky**
fixes https://github.com/G-Research/consuldotnet/issues/258
@marcin-krystianc @naskio

### CODE REVIEW

File: Consul.Test/AgentTest.cs [hunk 1]
```diff
@@ -993,5 +994,15 @@ public async Task Agent_FilterChecks()
             Assert.Single(actual);
             Assert.Equal(checkName, actual.Values.First().Name);
         }
+
+        [Fact]
+        public async Task Agent_HostInfo()
+        {
+            var agentVersion = await _client.Agent.GetAgentHostInfo();
```
[the remark this answers is not in the document]
Comment (sammychinedu2ky) [reply Consul.Test/AgentTest.cs:1001]: Makes sense

[the remark this answers is not in the document]
Comment (sammychinedu2ky) [reply Consul.Test/AgentTest.cs:1001]: Thanks

File: Consul.Test/AgentTest.cs [hunk 2]
```diff
@@ -993,5 +994,20 @@ public async Task Agent_FilterChecks()
...
+            var hostInfo = await _client.Agent.GetAgentHostInfo();
+            Assert.NotNull(hostInfo.Response.Host);
+            Assert.NotNull(hostInfo.Response.Memory);
+            Assert.NotNull(hostInfo.Response.CPU);
+            Assert.NotNull(hostInfo.Response.Disk);
+            Assert.True(hostInfo.Response.CollectionTime > 0);
+            Assert.True(agentVersion.Response.Host.Os.Contains("windows")
+                || agentVersion.Response.Host.Os.Contains("linux")
+                || agentVersion.Response.Host.Os.Contains("mac")
+                || agentVersion.Response.Host.Os.Contains("darwin"));
```
Comment (marcin-krystianc) [inline Consul.Test/AgentTest.cs:1010]:
```suggestion
```

**Comment by sammychinedu2ky** [comment]
So we are sticking to the set of listed os?

**Comment by marcin-krystianc** [comment]
> ?

No, I think it is not a good idea to do it. I forgot to include removal of it in the first patch.
BTW, there are conflicts to be resolved.

**Comment by sammychinedu2ky** [comment]
Okay 👌

**Comment by sammychinedu2ky** [comment]
@marcin-krystianc I've resolved the merge conflict. please can you check it again

**Review by marcin-krystianc — APPROVED** [review]

**Comment by sammychinedu2ky** [comment]
There seems to be some failed test

**Comment by marcin-krystianc** [comment]
> There seems to be some failed test

Yeah, they look legit. Can you try to investigate what is going on?

**Comment by sammychinedu2ky** [comment]
so let me leave it and work on the other code

File: Consul/Agent.cs [hunk 3]
```diff
@@ -430,6 +430,118 @@ public enum LogLevel
...
+    /// AgentHostInfo represents the host information for the agent
+    /// </summary>
+    public class AgentHostInfo
+    {
+        public MemoryInfo Memory { get; set; }
+        public List<CPUInfo> CPU { get; set; }
+        public HostInfo Host { get; set; }
+        public DiskInfo Disk { get; set; }
+        public long CollectionTime { get; set; }
+        public object Errors { get; set; }
```
Comment (marcin-krystianc) [inline Consul/Agent.cs:542]: I think that if we don't know what the type of `Error` is then, we should just remove it from the `AgentHostInfo` class.
It is better to not include it prematurely, as changing its type later would require to introduce a breaking change.

File: Consul.Test/AgentTest.cs [hunk 4]
```diff
@@ -994,6 +994,17 @@ public async Task Agent_FilterChecks()
             Assert.Equal(checkName, actual.Values.First().Name);
         }
 
+        [Fact]
+        public async Task Agent_HostInfo()
+        {
+            var hostInfo = await _client.Agent.GetAgentHostInfo();
+            Assert.NotNull(hostInfo.Response.Host);
+            Assert.NotNull(hostInfo.Response.Memory);
+            Assert.NotNull(hostInfo.Response.CPU);
```
Comment (marcin-krystianc) [inline Consul.Test/AgentTest.cs:1003]: Yeah, I tried it locally with version 1.9.17 and indeed, the CPU entry is missing in the repose. So lets remove the assertion for it.
```suggestion
```

**Comment by marcin-krystianc** [comment]
> so let me leave it and work on the other code

I think this change is still worth getting in. I doubt that there will be any major changes in the response body in the future.

**Comment by sammychinedu2ky** [comment]
I've done that.. once you approve of it, I would work on the merge conflicts

**Comment by sammychinedu2ky** [comment]
Ooh cool no conflicts 😄

**Comment by marcin-krystianc** [comment]
> Ooh cool no conflicts 😄

Yeah, it was good to merge. Thank you!

**Comment by sammychinedu2ky** [comment]
my pleasure!!

**Pull request merged**